Code/Framework/AzCore/AzCore/Math/RayIntersection.inl

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "AzCore/Math/Vector2.h"
+#include <AzCore/Math/MathUtils.h>
+#include <AzCore/Math/Ray.h>
+#include <AzCore/Math/Vector3.h>
+
+namespace AZ
+{
+    namespace RayIntersection
+    {
+        bool RayTriangle(
```

Comment:
update: I am sure this already exists in the code base, in fact I'm pretty sure @mbalfour-amzn implemented a very rigorous implementation of this. I'd please wait for his feedback before continuing with this PR

Reply:
Yep, I recently rewrote AZ::Intersect::IntersectSegmentTriangle() so that it uses a fast and accurate watertight algorithm. I'd prefer not to have multiple different versions of ray/segment to triangle intersection algorithms in the code. If there's a better version than the one we've got, we should replace it, otherwise please just use the existing one.

That said, this is also proposing a couple of differences to the API, since it uses a ray instead of a segment, and also returns the UV coordinates but _not_ the normal. If these are important API changes, I'd suggest creating a helper that basically does this:
* Convert ray to segment that goes from ray origin to max length in the direction of the ray.
* Calls IntersectSegmentTriangle()
* Calls AZ::Intersect::Barycentric() to get the UVW coordinates of the hit point.

I'm not sure the helper is actually what you want in this case though, since you're calling the methods from a tight loop that iterates across a bunch of triangles. You'd really instead want to do this in three parts, where you convert the ray once, create a SegmentTriangleHitTester so that you can minimize the calculations per triangle, and then only call Barycentric() once on the final successful result.